lib/ex_force.ex

```diff
@@ -350,34 +352,44 @@ defmodule ExForce do
 
   @spec start_query_stream(
           client,
-          (client, soql -> {:ok, QueryResult.t()} | any),
+          (client, soql -> {:ok, QueryResult.t()} | {:error, any}),
```

Comment:
I see… so the Stream is always returned, but the query doesn't happen until the first item is requested, potentially returning an error tuple.

Very nice!